cmake/compiler_flags_Clang_CXX.cmake

```diff
@@ -7,25 +7,25 @@
 # FLAGS COMMON TO ALL BUILD TYPES
 ####################################################################
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=gnu++14 -g -Wall -Wno-c++11-extensions")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14 -g -Wall -Wno-c++14-extensions")
```

Comment:
@climbfuji and @DJDavies2 was there a reason this was `gnu` a question asks

does this have an impact on anyone